.github/workflows/windows-syscollector-tests.yml

```diff
@@ -0,0 +1,114 @@
+name: Syscollector test on Windows
+
+on:
+  pull_request:
+    branches: ["*"]
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+      - name: Install mingw
+        run: sudo apt install gcc-mingw-w64 g++-mingw-w64-i686 g++-mingw-w64-x86-64 nsis -y
+      - name: make deps
+        run: make -C src deps TARGET=winagent -j2
+      - name: make
+        run: make -C src TARGET=winagent -j2
+        # Make folder and upload as artifacts.
+      - name: Copy libraries
+        run: |
+          mkdir -p src/shared_libraries
+          cp src/libwazuhext.dll src/shared_libraries/
+          cp src/win32/libgcc_s_dw2-1.dll src/shared_libraries/
+          cp src/win32/libstdc++-6.dll src/shared_libraries/
+          cp src/win32/libwinpthread-1.dll src/shared_libraries/
+      # Upload build artifacts for the shared libraries.
+      - name: Upload Artifact shared_libraries
+        uses: actions/upload-artifact@v2
+        with:
+          name: shared_libraries
+          path: src/shared_libraries
+      # Upload build artifacts for the data provider.
+      - name: Upload Artifact data_provider
+        uses: actions/upload-artifact@v2
+        with:
+          name: data_provider
+          path: src/data_provider/build/bin
+      # Upload build artifacts for dbsync.
+      - name: Upload Artifact dbsync
```

Comment:
Maybe `dbsync` and `rsync` are not required for running the test tool.

Reply:
we are going to get this ready to add new cases in a while soon.

This technical debt will be covered shortly.